### ISSUE TITLE
Global fallback for executing commands inside nonexistent directories (getcwd error)

### ISSUE THREAD

**Issue by xRiot** [clearlinux/distribution#1660]
This has been an issue since beginning of time with Linux. When executing certain commands inside non-existent directories, you'll usually get this error:
> getcwd: cannot access parent directories: No such file or directory

Swupd's command: `bundle-add {bundle}` is a good example of this issue:
```
(...)
shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory
(...)
job-working-directory: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory
(...)
```
**As it turns out this isn't easily patchable on application level** _(according to the [ticket #1078](https://github.com/clearlinux/swupd-client/issues/1078) I've submitted in swupd-client)_ and thus it would be a good time to consider a bigger fix, possibly in the kernel?

**Comment by ahkok** [comment]
The error messages are intentional... 

`bundle-add` causes various post update triggers to run, which are service units, each with a shell. Somewhere in that process a program executes and it performs checks on the working directory, and outputs the errors. This isn't a random error, someone wrote explicit code to display these messages because they thought it would be better to display them than to hide them as there are probably further consequences to be had.

**Comment by ahkok** [comment]
I don't believe that, from a fundamental standpoint, there is a problem here.

**Comment by xRiot** [comment]
This issue isn't specific to swupd at all @ahkok, it exists in a massive variety of packages. Having some sort of fallback in the kernel when executing commands in deleted directories would be very useful, although the proper implementation would need to be discussed.
